extensions/vault/deployment/src/main/java/io/quarkus/vault/deployment/DevServicesVaultProcessor.java

```diff
@@ -0,0 +1,202 @@
+package io.quarkus.vault.deployment;
+
+import java.io.Closeable;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.OptionalInt;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.jboss.logging.Logger;
+import org.testcontainers.utility.DockerImageName;
+import org.testcontainers.vault.VaultContainer;
+
+import io.quarkus.bootstrap.classloading.QuarkusClassLoader;
+import io.quarkus.deployment.IsDockerWorking;
+import io.quarkus.deployment.IsNormal;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.LaunchModeBuildItem;
+import io.quarkus.deployment.builditem.RunTimeConfigurationDefaultBuildItem;
+import io.quarkus.deployment.builditem.ServiceStartBuildItem;
+import io.quarkus.runtime.LaunchMode;
+import io.quarkus.runtime.configuration.ConfigUtils;
+import io.quarkus.vault.deployment.devservices.DevServicesVaultResultBuildItem;
+import io.quarkus.vault.runtime.config.DevServicesConfig;
+import io.quarkus.vault.runtime.config.VaultBuildTimeConfig;
+
+public class DevServicesVaultProcessor {
+    private static final Logger log = Logger.getLogger(DevServicesVaultProcessor.class);
+    private static final String VAULT_IMAGE = "vault:1.7.1";
```

Comment:
we should be consistent with `VaultTestExtension.getVaultImage()`. Let's upgrade everything to `1.7.1`.

Reply:
Added a `VaultVersions` class with the constant `VAULT_TEST_VERSION` in the `quarkus-vault` module. Both `DeviceServicesVaultProcessor` and `VaultTestExtensions` now select their version based on that constant.